test/app-luatest/gh_7000_compat_module_test.lua

```diff
@@ -0,0 +1,262 @@
+local t = require('luatest')
+local g = t.group()
+local compat = require('tarantool').compat
+
+local NEW = 'new'
+local OLD = 'old'
+local DEFAULT = 'default'
+
+local reset = function(compat)
+    local f = loadstring(compat.dump('default'))
+    f()
+end
+
+g.before_all(function() reset(compat) end)
+g.after_all( function() reset(compat) end)
+
+local option_1_called = false
+local option_2_called = false
+
+local hot_reload_option_called
+
+local policies = {
+    option_1_policy = {
+        name = 'option_1',
+        default = 'new',
+        brief = 'option_1',
+        action = function()
+            option_1_called = true
+        end,
+        run_action_now = true
+    },
+    option_2_policy = {
+        name = 'option_2',
+        default = 'old',
+        brief = 'option_2',
+        obsolete = nil,         -- Explicitly mark option as non-obsolete.
+        action = function()
+            option_2_called = true
+        end,
+        run_action_now = false
+  },
+}
+
+local obsolete_policy = {
+    name = 'obsolete_option',
+    default = 'new',
+    brief = 'obsolete_option',
+    obsolete = '5.0'
+  }
+
+local test_options_calls = function()
+    t.assert(option_1_called)
+    t.assert(option_2_called)
+    option_1_called = false
+    option_2_called = false
+end
+
+g.test_add_option = function()
```

Comment:
Other tests depend on this one. I mean, this doesn't works:

```shell
$ PATH="$(realpath src):${PATH}" luatest test/app-luatest/gh_7000_compat_module_test.lua -v -c -p test_serialize
```

While this works:

```shell
$ PATH="$(realpath src):${PATH}" luatest test/app-luatest/gh_7000_compat_module_test.lua -v -c -p test_add_option -p test_serialize
```

Can we overcome it in some not-very-complex way?

Reply:
We can move common `add_option()` to `before_all()` and add some other options in the `test_add_option`, but why would anyone run then separate?

Reply:
I often do it. I found this particular problem during review, because I added debug prints to show intermediate values to better understand what is going on. If others are working (or will work) with your testing code, it is good to allow them to understand the code piece-by-piece or, in other words, make test cases independent.

Reply:
Fixed.